tools/driver/p4c/main.py

```diff
@@ -0,0 +1,190 @@
+                        action="store_true", default=False)
+    parser.add_argument("-S", dest="run_till_assembler",
+                        help="Only run the preprocess and compilation steps",
+                        action="store_true", default=False)
+    parser.add_argument("-c", dest="run_all",
+                        help="Only run preprocess, compile, and assemble steps",
+                        action="store_true", default=True)
+    parser.add_argument("-x", dest="language",
+                        help="Treat subsequent input files as having type language.",
+                        action="store", default="p4-16")
+    parser.add_argument("-I", dest="search_path",
+                        help="Add directory to include search path",
+                        action="append", default=[])
+    parser.add_argument("-o", dest="output_directory",
+                        help="Write output to the provided path",
+                        action="store", metavar="PATH", default=".")
+
+    parser.add_argument("source_file", help="Files to compile")
+
+    # many more options
+    opts = parser.parse_args()
+    args = opts.source_file
+
+    if opts.show_version:
+        print("p4c %s" % (p4c.__version__))
+        sys.exit(0)
+
+    if not args:
+        parser.error('No input specified.')
+
+    init_globals()
+
+    save_options("preprocessor", opts.preprocessor_options)
+    save_options("compiler", opts.compiler_options)
+    save_options("assembler", opts.assembler_options)
+    save_options("linker", opts.linker_options)
+
+    # set output directory
+    if not os.path.exists(opts.output_directory):
+        os.makedirs(opts.output_directory)
+
+    if "P4C_INCLUDE_PATH" not in os.environ:
+        print "error: P4C_INCLUDE_PATH undefined."
+        sys.exit(1)
+
+    cfg_files = glob.glob("{}/*.cfg".format(os.path.dirname(__file__)))
+    cfg = p4c.config.Config(config_prefix = "p4c")
+    for cf in cfg_files:
+        cfg.load_from_config(cf, opts.output_directory, opts.source_file)
+
+
+    if opts.backend not in cfg.steps:
+        print "Unknown target:", opts.backend
+        sys.exit(1)
+
+    commands['preprocessor'].insert(0, cfg.get_preprocessor(opts.backend))
+    commands['compiler'].insert(0, cfg.get_compiler(opts.backend))
+    commands['assembler'].insert(0, cfg.get_assembler(opts.backend))
+    commands['linker'].insert(0, cfg.get_linker(opts.backend))
+
+    # handle mode flags
+    step_enable = [False, False, False, False]
+    if opts.run_preprocessor_only:
+        step_enable = [True, False, False, False]
+    elif opts.skip_preprocessor:
+        step_enable = [False, True, True, True]
+    elif opts.run_till_assembler:
+        step_enable = [True, True, False, False]
+    elif opts.run_all:
+        step_enable = [True, True, True, True]
+
+    # default search path
+    for path in os.environ['P4C_INCLUDE_PATH'].split(':'):
+        if (opts.language == 'p4-16' and os.path.isdir(path + '/p4include')):
```

Comment:
Given the way it is used, P4C_INCLUDE_PATH should probably be replaced by P4C_INSTALL_PREFIX, and should be specified .cfg file (possibly per target?) rather than in an envvar